toolchain/check/context.h

```diff
@@ -301,25 +301,23 @@ class Context {
   // If the type is not complete, `diagnoser` is invoked to diagnose the issue,
   // if a `diagnoser` is provided. The builder it returns will be annotated to
   // describe the reason why the type is not complete.
-  auto TryToCompleteType(
-      SemIR::TypeId type_id,
-      std::optional<BuildDiagnosticFn> diagnoser = std::nullopt) -> bool;
+  auto TryToCompleteType(SemIR::TypeId type_id,
+                         BuildDiagnosticFn diagnoser = nullptr) -> bool;
 
   // Attempts to complete and define the type `type_id`. Returns `true` if the
   // type is defined, or `false` if no definition is available. A defined type
   // has known members.
   //
   // This is the same as `TryToCompleteType` except for interfaces, which are
   // complete before they are fully defined.
-  auto TryToDefineType(
-      SemIR::TypeId type_id,
-      std::optional<BuildDiagnosticFn> diagnoser = std::nullopt) -> bool;
+  auto TryToDefineType(SemIR::TypeId type_id,
+                       BuildDiagnosticFn diagnoser = nullptr) -> bool;
 
   // Returns the type `type_id` as a complete type, or produces an incomplete
   // type error and returns an error type. This is a convenience wrapper around
   // TryToCompleteType.
-  auto AsCompleteType(SemIR::TypeId type_id, BuildDiagnosticFn diagnoser)
-      -> SemIR::TypeId {
+  auto AsCompleteType(SemIR::TypeId type_id,
+                      BuildDiagnosticFn diagnoser = nullptr) -> SemIR::TypeId {
```

Comment:
This should not have a default argument -- this diagnoser is intentionally not optional. If it were omitted, we'd return `Error` here without actually emitting an error diagnostic, which would be a bug.

Reply:
Ah, fair. Thanks for the catch/explanation. Fixed.

Reply:
(oops, forgot to push, done that now... )